batch/batch/aioclient.py

```diff
@@ -12,6 +12,27 @@
 
 
 class Job:
+    @staticmethod
+    def exit_code(job_status):
+        def _head(l):
+            return l[0] if len(l) != 0 else None
+
+        if 'exit_code' in job_status and job_status['exit_code'] is not None:
+            exit_codes = job_status['exit_code']
+            exit_codes = [exit_codes.get(task) for task in ['input', 'main', 'output']]
+
+            any_missing_ec = any([ec is None for ec in exit_codes])
+            first_nonzero_ec = _head([ec for ec in exit_codes if ec is not None and ec != 0])
+            all_zero_ec = all([ec is not None and ec == 0 for ec in exit_codes])
+
+            if any_missing_ec:
+                return None
+            elif all_zero_ec:
+                return 0
+            else:
+                return first_nonzero_ec
```

Comment:
If there is a nonzero, it will always be the exit code, so you can return it.  (Interestingly, you can know the exit code before the job is finished.)  I think you can do it like this:

```
  i = 0
  while i < len(exit_codes):
    ec = exit_codes[i]
    if ec > 0:
      return ec
    if ec is None:
      return None
    i += 1
  return 0
```

Reply:
Seems right to me. Fixed